src/SFA.DAS.EmployerApprenticeshipsService.Infrastructure/Services/ExcelService.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using ClosedXML.Excel;
+using SFA.DAS.EAS.Domain.Interfaces;
+
+namespace SFA.DAS.EAS.Infrastructure.Services
+{
+    public class ExcelService : IExcelService
+    {
+        public Dictionary<string, string[][]> ReadExcelFile(byte[] fileData)
+        {
+            var workbookData = new Dictionary<string, string[][]>();
+
+            using (var stream = new MemoryStream(fileData))
```

Comment:
There is a lot of nesting going on here - could it be simplified?

Reply:
it was change to increase performance as I had some issues with the old design that was a little less nested. The library i'm using only seems to allow cell by cell edits so by the nature of the importing it falls into a nested approach workbook -> worksheet -> row -> column. The outer nesting is handling the dispose aspects of the data handling objects. It fairly simple as it is but i could move some of the code to different methods if you think that would be useful?

Reply:
I personally would try to simplify it but if you're happy then that's cool - just leave it.